string-manipulation-2/src/swap-chars.js

```diff
@@ -0,0 +1,31 @@
+/* exported swapChars */
+
+// loop over the characters
+// slice at each character
+// create an empty array to push each character
+// if the index value is the firstIndex, save it in a variable
+// push a space character as placeholder
+// if the index value is the secondIndex, save it in a variable
+// push a space character as placeholder
+// if the index value is not the firstIndex or the secondIndex push the character at the index
+// reassign the value of character at second index of array to the firstChar variable
+// reassign the value of character at first index of array to the secondChar variable
+// join the return
+
+function swapChars(firstIndex, secondIndex, string) {
+  var firstChar = string[firstIndex];
+  var secondChar = string[secondIndex];
+  var newString = [];
+  for (var i = 0; i < string.length; i++) {
+    if (i === firstIndex) {
+      newString.push(' ');
+    } else if (i === secondIndex) {
+      newString.push(' ');
+    } else {
+      newString.push(string.slice(i, i + 1));
+    }
+  }
+  newString[secondIndex] = firstChar;
+  newString[firstIndex] = secondChar;
+  return newString.join('');
```

Comment:
`join` does a full loop over the array to make the string. You should concatenate together your output rather than use an array, which will cut out an extra loop.